test/1_stdlib/StringDiagnostics_without_Foundation.swift

```diff
@@ -8,13 +8,47 @@ func acceptsRandomAccessCollection<I: RandomAccessCollection>(_: I) {}
 func testStringCollectionTypes(s: String) {
   acceptsCollection(s.utf8)
   acceptsBidirectionalCollection(s.utf8) // expected-error{{argument type 'String.UTF8View' does not conform to expected type 'BidirectionalCollection'}}
+  acceptsRandomAccessCollection(s.utf8) // expected-error{{argument type 'String.UTF8View' does not conform to expected type 'RandomAccessCollection'}}
+
+  // UTF16View is random-access with Foundation, bidirectional without
+  acceptsCollection(s.utf16)
+  acceptsBidirectionalCollection(s.utf16)
+  acceptsRandomAccessCollection(s.utf16) // expected-error{{argument type 'String.UTF16View' does not conform to expected type 'RandomAccessCollection'}}
+
+  acceptsCollection(s.unicodeScalars)
   acceptsBidirectionalCollection(s.unicodeScalars)
   acceptsRandomAccessCollection(s.unicodeScalars) // expected-error{{argument type 'String.UnicodeScalarView' does not conform to expected type 'RandomAccessCollection'}}
+
+  acceptsCollection(s.characters)
   acceptsBidirectionalCollection(s.characters)
   acceptsRandomAccessCollection(s.characters) // expected-error{{argument type 'String.CharacterView' does not conform to expected type 'RandomAccessCollection'}}
-  
-  // UTF16View is random-access with Foundation, bidirectional without
-  acceptsBidirectionalCollection(s.utf16)
-  acceptsRandomAccessCollection(s.utf16) // expected-error{{argument type 'String.UTF16View' does not conform to expected type 'RandomAccessCollection'}}
 }
 
+// Positive and negative tests for String index types. Testing the complete
+// set of conformance just in case protocol hierarchy changes.
+func acceptsCollectionIndex<I: Comparable>(_: I) {}
```

Comment:
I think these index tests (from here to the end of the file) can be dropped if you implement the `expectCollectionAssociatedTypes` test.